src/nashpy/learning/replicator_dynamics.py

```diff
@@ -55,6 +55,10 @@ def asymmetric_replicator_dynamics(A, B, x0=None, y0=None, timepoints=None):
     """
     Implement asymmetric replicator dynamics
     """
+
+    if not A.shape == B.shape:
+        raise ValueError("Unequal dimensions for matrices A and B")
```

Comment:
This probably makes more sense to be the `__init__` of the game class (if you could pop it there, that would be great -- needs a test if you do).

Reply:
You already have the test I see so yeah if you could just pop that in the `__init__` for the game that would be great as the mismatch of the dimensions doesn't make sense for any other algorithm.